pandas/tests/groupby/test_grouping.py

```diff
@@ -0,0 +1,732 @@
+# -*- coding: utf-8 -*-
+
+""" test where we are determining what we are grouping, or getting groups """
+
+import pytest
+
+from warnings import catch_warnings
+from pandas import (date_range, Timestamp,
+                    Index, MultiIndex, DataFrame, Series)
+from pandas.util.testing import (assert_panel_equal, assert_frame_equal,
+                                 assert_series_equal, assert_almost_equal)
+from pandas.compat import lrange
+
+from pandas import compat
+import numpy as np
+
+import pandas.util.testing as tm
+import pandas as pd
+from .common import MixIn
+
+
+# selection
+# --------------------------------
+
+class TestSelection(MixIn):
+
+    def test_select_bad_cols(self):
+        df = DataFrame([[1, 2]], columns=['A', 'B'])
+        g = df.groupby('A')
+        pytest.raises(KeyError, g.__getitem__, ['C'])  # g[['C']]
+
+        pytest.raises(KeyError, g.__getitem__, ['A', 'C'])  # g[['A', 'C']]
+        with tm.assert_raises_regex(KeyError, '^[^A]+$'):
```

Comment:
On a somewhat relevant point to my comment in <a href="https://github.com/pandas-dev/pandas/pull/18047#issuecomment-340659137">#18047</a>, perhaps we need a built-in way to implement the `assert not` checks so that we avoid ugly regex like this.